### PR TITLE
[4.1] Add net8.0-windows10.0.19041.0 Targets to Uno

### PR DESCRIPTION
**Pull request by inforithmics**
Please summarize your PR and explain which problem it solves.

Fixes this: https://github.com/Mapsui/Mapsui/issues/2523

### CODE REVIEW

File: Samples/Mapsui.Samples.Uno/Mapsui.Samples.Uno.Skia.Gtk/Program.cs [hunk 1]
```diff
@@ -14,7 +15,7 @@ static void Main(string[] args)
             expArgs.ExitApplication = true;
         };
 
-        var host = new GtkHost(() => new App(), args);
```
Comment (pauldendulk) [inline Samples/Mapsui.Samples.Uno/Mapsui.Samples.Uno.Skia.Gtk/Program.cs:17]: 👍

File: Directory.Packages.props [hunk 2]
```diff
@@ -107,23 +107,23 @@
     <PackageVersion Include="Uno.Extensions.Logging.WebAssembly.Console" Version="[1.4.0,2.0.0)" />
     <PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="[4.0.1,5.0.0)" />
     <PackageVersion Include="Uno.UniversalImageLoader" Version="[1.9.36,2.0.0)" />
-    <PackageVersion Include="Uno.UI" Version="[5.0.19,6.0.0)" />
-    <PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="[5.0.19,6.0.0)" />
-    <PackageVersion Include="Uno.UI.RemoteControl" Version="[5.0.19,6.0.0)" />
+    <PackageVersion Include="Uno.UI" Version="[5.1.31,6.0.0)" />
```
Comment (pauldendulk) [inline Directory.Packages.props:110]: I wonder how this affects uses on a lower version. Still, this needs to be done.

File: Mapsui.NoMSBuild.slnf [hunk 3]
```diff
@@ -37,6 +37,7 @@
       "Samples\\Mapsui.Samples.Maui\\Mapsui.Samples.Maui.csproj",
       "Samples\\Mapsui.Samples.Uno.WinUI\\Mapsui.Samples.Uno.WinUI.Mobile\\Mapsui.Samples.Uno.WinUI.Mobile.csproj",
       "Samples\\Mapsui.Samples.Uno.WinUI\\Mapsui.Samples.Uno.WinUI.Skia.Gtk\\Mapsui.Samples.Uno.WinUI.Skia.Gtk.csproj",
+      "Samples\\Mapsui.Samples.Uno.WinUI\\Mapsui.Samples.Uno.WinUI.Wasm\\Mapsui.Samples.Uno.WinUI.Wasm.csproj",
```
Comment (pauldendulk) [inline Mapsui.NoMSBuild.slnf:40]: Great wasm is back now.

**Review by pauldendulk — APPROVED** [review]

File: Mapsui.UI.Uno.WinUI/Mapsui.UI.Uno.WinUI.csproj [hunk 4]
```diff
@@ -37,7 +37,7 @@
   </PropertyGroup>
 
   <ItemGroup Condition="$(TargetFramework.Contains('windows'))">
-    <PackageReference Include="Microsoft.WindowsAppSDK" VersionOverride="[1.3.230502000,2.0.0)"  />
+    <PackageReference Include="Microsoft.WindowsAppSDK" VersionOverride="[1.5.240227000,2.0.0)"  />
```
[the remark this answers is not in the document]
Comment (pauldendulk) [reply Mapsui.UI.Uno.WinUI/Mapsui.UI.Uno.WinUI.csproj:40]: I understand that this is needed to fix .NET 8, but I am talking about a problem resulting from this, and that is that this is a breaking change (I think) for users that are on appSDK 1.3. So, if they just update a patch version their app does not work anymore. They have reason to be upset by this. It violates semver. I think we have to do this given the current situation, but we might have to thing about a better strategy. So, there are two things I am thinking about:

- Is it really a breaking change? And how many users are affected by it?
- How could we avoid this problem? Perhaps we should not have tried to add .NET 8 to v4 but have a v5 release which has only .NET update. Perhaps we should align our release cadence with the .NET releases.

[the remark this answers is not in the document]
Comment (inforithmics) [reply Mapsui.UI.Uno.WinUI/Mapsui.UI.Uno.WinUI.csproj:40]: Then I'll make a pull request to lower the version again to 1.3.

File: Mapsui.sln.DotSettings [hunk 5]
```diff
@@ -224,6 +224,7 @@
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=XY/@EntryIndexedValue">XY</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/PredefinedNamingRules/=MethodPropertyEvent/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb_AaBb" /&gt;</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/PredefinedNamingRules/=PrivateInstanceFields/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;</s:String>
+	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/UserRules/=4a98fdf6_002D7d98_002D4f5a_002Dafeb_002Dea44ad98c70c/@EntryIndexedValue">&lt;Policy&gt;&lt;Descriptor Staticness="Instance" AccessRightKinds="Private" Description="Instance fields (private)"&gt;&lt;ElementKinds&gt;&lt;Kind Name="FIELD" /&gt;&lt;Kind Name="READONLY_FIELD" /&gt;&lt;/ElementKinds&gt;&lt;/Descriptor&gt;&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;&lt;/Policy&gt;</s:String>
```
[the remark this answers is not in the document]
Comment (pauldendulk) [reply Mapsui.sln.DotSettings:227]: oh, this is v4. They were already removed in main.

File: Mapsui.UI.Uno.WinUI/Mapsui.UI.Uno.WinUI.csproj (hunk 4 above)
[the remark this answers is not in the document]
Comment (pauldendulk) [reply Mapsui.UI.Uno.WinUI/Mapsui.UI.Uno.WinUI.csproj:40]: Please don't. 

I am not criticizing this pull request but I am thinking about the problem in general. I I don't know what the best solution is. Perhaps it was a mistake to try to support .NET 8 and .NET 7 in the same version. Perhaps we should have separate release for all head packages (I don't like the maintenance which comes this, I like the simplicity of the current releases). Perhaps we should drop semver. Perhaps we should update the major version.

**Pull request merged**